### PR TITLE
fix errors on create user on Launch

### PR DESCRIPTION
**Pull request by LeonidMorozov**

### CODE REVIEW

File: app/controllers/lti_provider/lti_controller.rb [hunk 1]
```diff
@@ -101,14 +103,21 @@ def allow_iframe
 
       def new_user_params(app, username, provider_params)
         unless @user_params
-          @user_params = {provider: app.name, promo: app.name, username: username}
+          @user_params = {provider: app.name, promo: app.name}
+          @user_params[:username] = username if username.present?
           @user_params[:password_confirmation] = @user_params[:password] = SecureRandom.hex
```
Comment (mzverev) [inline app/controllers/lti_provider/lti_controller.rb]: People with usernames (no email) won't be able to change the password. I suppose we need to make sure that username (not email) always comes with the password.

**Comment by mzverev** [comment]
@LeonidMorozov : check my comment

File: app/controllers/lti_provider/lti_controller.rb (hunk 1 above)
Comment (LeonidMorozov) [inline app/controllers/lti_provider/lti_controller.rb]: @mzverev , I agree.
But here's corner-case. LTI may not return email and username at all, just user_id in LMS.
So, here's 2 ways:
  - allow anonymous Assessment/Homework Run at least for LTI runs
  - find/create account on any Assessment/Homework Run

We decided to use 2nd way today. Ideally user should be created by silen_login before LTI run, so, it should be on our system.

Comment (mzverev) [inline app/controllers/lti_provider/lti_controller.rb]: @LeonidMorozov : please create a story to NOT to create managed users without password. Either email or password must be present.

**Pull request merged**